backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,230 @@
+                string errorMessage = $"The robot {robot.Name} is on installation {robotInstallation.Name} which is not the same as the mission installation {missionInstallation.Name}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new MissionException(errorMessage);
+            }
+        }
+
+        public async Task EnsureRobotIsCorrectlyLocalized(Robot robot, MissionDefinition missionDefinition)
+        {
+            if (missionDefinition.Area is null)
+            {
+                string errorMessage = $"There was no area associated with mission definition {missionDefinition.Id}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new AreaNotFoundException(errorMessage);
+            }
+
+            if (!await RobotIsLocalized(robot.Id)) { await StartLocalizationMissionInArea(robot, missionDefinition.Area); }
+
+            if (!RobotIsOnSameDeckAsMission(robot.CurrentArea, missionDefinition.Area))
+            {
+                string errorMessage = $"A new mission run of mission definition {missionDefinition.Id} will not be started as the robot is not localized on the same deck as the mission";
+                _logger.LogError("{Message}", errorMessage);
+                throw new RobotLocalizationException(errorMessage);
+            }
+        }
+
+        public async Task<bool> RobotIsLocalized(string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot != null)
+            {
+                return robot.CurrentArea is not null;
+            }
+
+            string errorMessage = $"Robot with ID: {robotId} was not found in the database";
+            _logger.LogError("{Message}", errorMessage);
+            throw new RobotNotFoundException(errorMessage);
+        }
+
+        public async Task EnsureRobotWasCorrectlyLocalizedInPreviousMissionRun(string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                string errorMessage = $"Robot with ID: {robotId} was not found in the database";
+                _logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            if (await _missionRunService.OngoingMission(robot.Id))
+            {
+                await WaitForLocalizationMissionStatusToBeUpdated(robot);
+            }
+
+            var lastExecutedMissionRun = await _missionRunService.ReadLastExecutedMissionRunByRobot(robot.Id);
+            if (lastExecutedMissionRun is null)
+            {
+                string errorMessage = $"Could not find last executed mission run for robot with ID {robot.Id}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new MissionNotFoundException(errorMessage);
+            }
+
+            if (lastExecutedMissionRun.Status != MissionStatus.Successful)
```

Comment:
This means that if the previous mission was stopped (cancelled), for example, all the subsequent missions will be cancelled, is that intended behaviour?